### PR TITLE
[Merged by Bors] - Replace ganache-cli with anvil

### PR DESCRIPTION
**Pull request by pawanjay176**
## Issue Addressed

N/A

## Proposed Changes

Replace ganache-cli with anvil https://github.com/foundry-rs/foundry/blob/master/anvil/README.md
We can lose all js dependencies in CI as a consequence.

## Additional info
Also changes the ethers-rs version used in the execution layer (for the transaction reconstruction) to a newer one. This was necessary to get use the ethers utils for anvil. The fixed execution engine integration tests should catch any potential issues with the payload reconstruction after #3592

### CODE REVIEW

**Comment by michaelsproul** [comment]
Just came across this again and I'm very keen to adopt it because it seems like it would allow us to remove the dependency on the `web3` crate in favour of `ethers`. The `web3` crate is the source of many outdated and unmaintained deps in our `Cargo.lock`.

**Comment by pawanjay176** [comment]
@michaelsproul Fixed this up. I had to use the downgraded versions of ethers in the `eth1_test_rig` to satisfy msrv requirements. 
Using `cargo install` for anvil in CI but we could also potentially look into https://github.com/foundry-rs/foundry-toolchain to speed up `anvil` installation a bit.

**Comment by michaelsproul** [comment]
@pawanjay176 we're about to bump our MSRV to 1.65 for Capella anyway, so I think we may as well bump it in this PR and stick to one version of `ethers`?

I love seeing the `Cargo.lock` with no `web3` crate 😍. Will be awesome if we can make it a net reduction in number of deps by consolidating the `ethers` version too!

**Comment by pawanjay176** [comment]
Bumped up the ethers version. As discussed, we could merge it when we merge capella where msrv is already 1.65.
Blocked until then.

**Comment by pawanjay176** [comment]
Unblocking this guy. cc @michaelsproul

File: Cargo.lock
Comment (michaelsproul) [inline Cargo.lock]: Ah looks like my arbitrary hacks are broken because of the latest release of `derive_arbitrary`. Unfortunately we still need a hacky version because of this issue: https://github.com/rust-lang/rust-clippy/issues/10185

I'll cook up a new version

Comment (pawanjay176) [reply Cargo.lock]: Sorry seems like I had committed a stale lockfile. Pushed a new commit now

**Comment by michaelsproul** [comment]
Sorry I pushed another commit to mess with arbitrary, try merging unstable again and then we can merge :pray:

**Comment by pawanjay176** [comment]
Merged unstable @michaelsproul

File: testing/eth1_test_rig/src/lib.rs
Comment (michaelsproul) [inline testing/eth1_test_rig/src/lib.rs]: could propagate these errors

**Review by michaelsproul — APPROVED** [review]
Approved modulo one small error fix

**Comment by paulhauner** [comment]
I flagged this as ready-for-review since that seems appropriate given the last few comments :)

File: Cargo.lock
Comment (michaelsproul) [inline Cargo.lock:9771]: this looks really weird, but it's because of snow:

https://github.com/mcginty/snow/releases/tag/v0.9.1

**Review by michaelsproul — APPROVED** [review]

**Pull request closed without merging**
